File: src/test/java/com/christophsens/cvenhancer/prompt/TranslationPromptTest.java
package com.christophsens.cvenhancer.prompt;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.ai.chat.prompt.Prompt;

public class TranslationPromptTest {

    @Test
    void testTranslationPrompt() {
        String expectedResult = """
        Translate the text according to iso693 "en_GB"
        This is the cv: "cv"
        """;
        Prompt prompt = TranslationPrompt.getPrompt("cv", "en", "GB");
        String result = prompt.getContents();
        Assertions.assertEquals(expectedResult, result);
    }
}

File: src/test/kotlin/com/christophsens/cvenhancer/prompt/TranslationPromptTest.kt
package com.christophsens.cvenhancer.prompt

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class TranslationPromptTest {

    @Test
    fun testGetPrompt() {
        val cv = "Sample CV content"
        val languageCode = "en"
        val countryCode = "US"
        val contents = TranslationPrompt.getPrompt(cv, languageCode, countryCode).contents

        assertEquals("Translate the text according to iso693 \"en_US\"\nThis is the cv: \"${cv}\"", contents)
    }
}